apps/air-discount-scheme/backend/src/app/modules/nationalRegistry/nationalRegistry.service.ts

```diff
@@ -194,13 +196,25 @@ export class NationalRegistryService {
       .get(`${this.baseUrl}/family-lookup?ssn=${nationalId}`)
       .toPromise()
 
-    const children = response.data[0].results
-      .filter((person) => person.ssn !== nationalId && this.isChild(person.ssn))
-      .map((person) => person.ssn)
-    if (children) {
-      await this.cacheManager.set(cacheKey, { children }, { ttl: ONE_MONTH })
+    const data = response.data[0]
+    if (data.error) {
+      this.logger.error(`Could not find family members for User<${nationalId}>`)
+      return []
     }
 
+    const family = data.results
+    const user = family.find((person) => person.ssn === nationalId)
+    let children = []
+    if ([1, 2].includes(user.gender)) {
+      children = family
+        .filter(
+          (person) => person.ssn !== nationalId && this.isChild(person.ssn),
```

Comment:
Should we add to the `isChild` function to check gender, that it isn't 1 or 2.
We could maybe have a parent which is 17

Reply:
Yeah good point!